app/src/main/java/org/coepi/android/ui/symptoms/SymptomsFragment.kt

```diff
@@ -37,4 +38,15 @@ class SymptomsFragment : Fragment() {
             adapter.submitList(it.toMutableList())
         }
     }.root
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        viewModel.isInProgress.observe(viewLifecycleOwner, onChanged = {
+            if(it) {
+                this.view?.isClickable = false
+            }
+        })
+
+    }
```

Comment:
Let's try out a more lightweight approach:

- The loading view should always consume touch when present. Don't remember if it's possible to configure this via only XML attributes. In worst case, implement it as a custom view.

- When there's no active loading, the view's visibility should be `GONE` (I believe this to be already the case), so it will not consume touch.

This way we don't have to change anything in code.

Reply:
Gotcha, I'll look into implementing this in XML!